conjure-rust/src/main.rs

```diff
@@ -13,63 +13,61 @@
 // limitations under the License.
 #![warn(clippy::all)]
 
+use clap::{AppSettings, Parser};
 use std::path::PathBuf;
 use std::process;
-use structopt::clap::AppSettings;
-use structopt::StructOpt;
 
-#[derive(StructOpt)]
+#[derive(Parser)]
 enum Opts {
-    #[structopt(
+    #[clap(
         name = "generate",
-        setting = AppSettings::UnifiedHelpMessage,
+        dont_collapse_args_in_usage = true,
         setting = AppSettings::DeriveDisplayOrder,
-        setting = AppSettings::DontCollapseArgsInUsage,
     )]
     /// Generate Rust code from a conjure IR file.
     Generate(Args),
 }
 
-// FIXME move aliases over to the standard names
-#[derive(StructOpt)]
+// FIXME remove aliases
+#[derive(Parser)]
 struct Args {
-    #[structopt(long = "exhaustive")]
+    #[clap(long = "exhaustive")]
     /// Generate exhaustively matchable enums and unions
     exhaustive: bool,
-    #[structopt(long = "use-staged-builders", alias = "useStagedBuilders")]
+    #[clap(long = "useStagedBuilders", alias = "use-staged-builders")]
     /// Generate compile-time safe builders to ensure all required attributes are set
     use_staged_builders: bool,
-    #[structopt(long = "strip-prefix", value_name = "prefix", alias = "stripPrefix")]
+    #[clap(long = "stripPrefix", value_name = "prefix", alias = "strip-prefix")]
     /// Strip a prefix from types's package paths
     strip_prefix: Option<String>,
     /// The name of the generated crate
-    #[structopt(
-        long = "crate-name",
+    #[clap(
+        long = "productName",
```

Comment:
Why are we changing `long` from `crate-name` to `productName`? Same question below for version.

Reply:
We originally used `kebab-case` names for the command line options, but the Conjure CLI spec mandates `camelCase` (I think to more easily configure things in gradle-conjure?). We've supported both for a long time via aliases, but this just switches the defaults around so the `kebab-case` names are the aliases while the `camelCase` names are the normal ones that show up in `--help` etc.